From d2e39d4dc246f6df91c2d663471bfd18b72f0029 Mon Sep 17 00:00:00 2001
From: Chris Hofer <christian.hofer@codasip.com>
Date: Tue, 29 Jul 2025 17:03:45 +0200
Subject: [PATCH 3/6] tests/19.t: Unify io_event ring buffer layout

Test 19 reads an io_event directly from the kernel's ring buffer. For
this to work, the test and the kernel have to have the same memory
layout. So change the test's local definition to that of the kernel.

Upstream-Status: Inappropriate [CHERI-specific]
Signed-off-by: Chris Hofer <christian.hofer@codasip.com>
---
 harness/cases/19.t | 24 ++++++++++++++----------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/harness/cases/19.t b/harness/cases/19.t
index ba1c620..09d32ff 100644
--- a/harness/cases/19.t
+++ b/harness/cases/19.t
@@ -22,16 +22,20 @@
 volatile sig_atomic_t timed_out = 0;
 
 struct aio_ring {
-	unsigned	id;	/* kernel internal index number */
-	unsigned	nr;	/* number of io_events */
-	volatile unsigned	head;
-	volatile unsigned	tail;
-
-	unsigned	magic;
-	unsigned	compat_features;
-	unsigned	incompat_features;
-	unsigned	header_length;	/* size of aio_ring */
-
+	union {
+		struct {
+			unsigned	id;	/* kernel internal index number */
+			unsigned	nr;	/* number of io_events */
+			volatile unsigned	head;
+			volatile unsigned	tail;
+
+			unsigned	magic;
+			unsigned	compat_features;
+			unsigned	incompat_features;
+			unsigned	header_length;	/* size of aio_ring */
+		};
+		struct io_event pad;
+	};
 	struct io_event io_events[0];
 };
 
-- 
2.47.1

